src/script/api/script_order.cpp

```diff
@@ -515,18 +519,18 @@ static ScriptOrder::OrderPosition RealOrderPositionToScriptOrderPosition(Vehicle
 			order.SetLoadType((OrderLoadFlags)GB(order_flags, 5, 3));
 			order.SetUnloadType((OrderUnloadFlags)GB(order_flags, 2, 3));
 			order.SetStopLocation(OSL_PLATFORM_FAR_END);
+			if (v->IsGroundVehicle()) order.SetNonStopType((OrderNonStopFlags)GB(order_flags, 0, 2));
 			break;
 
 		case OT_GOTO_WAYPOINT:
 			order.MakeGoToWaypoint(::GetStationIndex(destination));
+			if (v->type == VEH_TRAIN) order.SetNonStopType((OrderNonStopFlags)GB(order_flags, 0, 2));
```

Comment:
```suggestion
			if (v->IsGroundVehicle()) order.SetNonStopType((OrderNonStopFlags)GB(order_flags, 0, 2));
```
This probably should be changed in the future when road waypoints are properly supported (#13398, #13419 and maybe some others...)